staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -277,6 +277,19 @@ func IsHTTPHeaderName(value string) []string {
 	return nil
 }
 
+const envVarNameFmt = "[-._a-zA-Z][-._a-zA-Z0-9]*"
```

Comment:
I can't fully judge on that, though applications parsing such environment variables should not have any problems (if they decide to support names with leading dots). FWIW, ConfigMaps with leading dots are [still forbidden](https://github.com/kubernetes/kubernetes/blob/d86c267112f17cd0ded06c44db9308dd3375a553/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L294-L297).

Would you like me to keep disallowing such names?